### PR TITLE
[styles] Reduce Tokyo subway visibility from z12- to z14-

### PR DESCRIPTION
**Pull request by charlie2clarke**
Edits and discussions referenced in #5837

This PR proposes reducing the current visibility of Tokyo subway entrances **from this**:
z12

**To this**:
z12

### CODE REVIEW

**Comment by biodranik** [comment]
Should toll booths be visible at that zoom level in a non-vehicle style?

@pastk does it make sense to switch to the vehicle style when users are planning the driving route?

**Comment by map-per** [comment]
> [styles] Reduce Tokyo subway entrance visibility from z12 to z13

You reduced the visibility to z14, or?

**Comment by charlie2clarke** [comment]
> Should toll booths be visible at that zoom level in a non-vehicle style?

Would there be a reason for someone wanting to see toll booths when not driving?

File: data/styles/clear/include/Subways.mapcss
Comment (charlie2clarke) [inline data/styles/clear/include/Subways.mapcss:1]: @map-per Re: https://github.com/organicmaps/organicmaps/pull/6291#issuecomment-1759046589 - icons would only be visible from z13 and above

**Comment by pastk** [comment]
> @pastk does it make sense to switch to the vehicle style when users are planning the driving route?

IDK, the roads in the vehicle style are too thick probably..
Just try to use it yourself for planning via `?vlight` switch.

The tollbooths were z12- for a long time AFAIK. Maybe it was assumed they're quite rare *inside* of the cities.

**Review by pastk — COMMENTED** [review]
Your change actually reduces visibility from z12- to z14- (which looks good for the Tokyo case).
But please update the commit and PR messages.

Also, could you please do the same change for the vehicle style?

**Comment by pastk** [comment]
Did you close it by accident?

**Comment by charlie2clarke** [comment]
> Did you close it by accident?

Yes, sorry

File: data/styles/vehicle/include/Subways.mapcss [hunk 1]
```diff
@@ -842,7 +842,7 @@ node|z17-[railway=station][transport=subway][city=tianjin] {icon-image: subway-t
 node|z16-[railway=subway_entrance][city=tianjin] {icon-image: subway-tianjin-s.svg;}
 
 /* Tokyo Subway Station */
-node|z13[railway=station][transport=subway][city=tokyo] {icon-image: subway-tokyo-s.svg;icon-min-distance: 1;}
+node|z13[railway=station][transport=subway][city=tokyo] {icon-image: none;}
```
Comment (pastk) [inline data/styles/vehicle/include/Subways.mapcss:845]: please add `text: none;` here also
default subway style in the vehicle mode has a label at z13 - we don't want to display a label without an icon

**Review by pastk — COMMENTED** [review]
Looks good, except the last glitch:

File: data/styles/vehicle/include/Subways.mapcss (hunk 1 above)
Comment (charlie2clarke) [reply data/styles/vehicle/include/Subways.mapcss]: Done. I will do some reading for my next change to styles to get to bottom of how everything works

File: data/styles/vehicle/include/priorities_4_overlays.prio.txt [hunk 2]
```diff
@@ -410,7 +410,7 @@ railway-station-subway-tashkent                     # icon z13- (also has captio
 railway-station-subway-tbilisi                      # icon z13- (also has caption(optional) z13-)
 railway-station-subway-tehran                       # icon z13- (also has caption(optional) z13-)
 railway-station-subway-tianjin                      # icon z13- (also has caption(optional) z13-)
-railway-station-subway-tokyo                        # icon z13- (also has caption(optional) z13-)
+railway-station-subway-tokyo                        # icon z14- and caption z13 (also has caption(optional) z14-)
```
Comment (pastk) [inline data/styles/vehicle/include/priorities_4_overlays.prio.txt:413]: I see your `text: none;` addition to the Subways.mapcss, but I see no changes in the generated files (still shows a caption at z13). Did you regenerate them?

Comment (charlie2clarke) [reply data/styles/vehicle/include/priorities_4_overlays.prio.txt:413]: Yeah I realised that so I put the PR as a draft. Will update with (hopefully) the last needed change shortly.

**Comment by charlie2clarke** [comment]
When setting 'text: none;' on the vehicle Subways.mapcss generating drules gives the following error:
`ERROR: priority is not set for caption railway-station-subway-tokyo::int_name`

**Comment by charlie2clarke** [comment]
Current style:
tokyo

File: data/styles/vehicle/include/Subways.mapcss [hunk 3]
```diff
@@ -842,7 +842,8 @@ node|z17-[railway=station][transport=subway][city=tianjin] {icon-image: subway-t
 node|z16-[railway=subway_entrance][city=tianjin] {icon-image: subway-tianjin-s.svg;}
 
 /* Tokyo Subway Station */
-node|z13[railway=station][transport=subway][city=tokyo] {icon-image: subway-tokyo-s.svg;icon-min-distance: 1;}
+node|z13[railway=station][transport=subway][city=tokyo]::int_name,
+{icon-image: none;text: none;}
```
Comment (pastk) [inline data/styles/vehicle/include/Subways.mapcss:846]: As you can see in `drules_proto_vehicle_clear.txt` diff, it removed only the secondary label (which is declared via ::int_name) and reset the icon to a generic one.
The right solution would be 
```
node|z13[railway=station][transport=subway][city=tokyo] {icon-image: none; text: none;}
node|z13[railway=station][transport=subway][city=tokyo]::int_name {text: none;}
```
First line removes the icon and primary label, second line removes the secondary label.

Comment (pastk) [reply data/styles/vehicle/include/Subways.mapcss:846]: Also note that you can test vehicle style changes via typing `?vlight` in the search to switch the style and `?light` to switch it back to the main style.

**Comment by charlie2clarke** [comment]
Are you happy with this styling then?

For normal:
normal

For vehicle:
normal

**Review by pastk — APPROVED** [review]
LGTM!
@vng

**Pull request merged**